File: Backend/Entities/Student.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Entities
{
    /// <summary>A class for representing a student.</summary>
    [Table("Students")]
    public class Student : DomainUser
    {
        public int EntranceYear { get; set; } // might be redundant

        public DepartmentInfo Major { get; set; }

        public ICollection<DepartmentInfo> Minors { get; set; }

        public double CGPA { get; set; }

        public double ExchangeScore { get; set; }

        public SemesterInfo PreferredSemester { get; set; }

        public ICollection<string> PreferredSchools { get; set; }

        public string ExchangeSchool { get; set; }

        public ICollection<CTEForm> CTEForms { get; set; }

        public ICollection<PreApprovalForm> PreApprovalForms { get; set; }

        public ICollection<EquivalenceRequest> EquivalenceRequestForms { get; set; }

        public ICollection<ToDoItem> ToDoList { get; set; } = new List<ToDoItem>();
    }
}

File: Frontend/src/app/_models/student.ts
import { DepartmentInfo } from './department-info';
import { SemesterInfo } from './semester-info';
import { CteForm } from './cte-form';
import { PreApprovalForm } from './pre-approval-form';
import { EquivalanceRequest } from './equivalance-request';
import { DomainUser } from './domain-user';

export interface Student extends DomainUser {
  entranceYear: number;
  major: DepartmentInfo;
  minors: DepartmentInfo[];
  cgpa: number;
  exchangeScore: number;
  preferredSemester: SemesterInfo;
  preferredSchools: string[];
  exchangeSchool: string;
  cteForms: CteForm[];
  preApprovalForms: PreApprovalForm[];
  equivalenceRequestForms: EquivalanceRequest[];
}
